rclcpp/test/test_node.cpp

```diff
@@ -73,6 +73,17 @@ TEST_F(TestNode, get_name_and_namespace) {
     EXPECT_STREQ("my_node", node->get_name());
     EXPECT_STREQ("/my/ns", node->get_namespace());
   }
+  {
+    auto node1 = std::make_shared<rclcpp::Node>("my_node1", "my/ns");
+    auto node2 = std::make_shared<rclcpp::Node>("my_node2", "my/ns");
+    auto node3 = std::make_shared<rclcpp::Node>("my_node3", "/ns2");
+    auto node4 = std::make_shared<rclcpp::Node>("my_node4", "my/ns3");
+    auto names_and_namespaces = node1->get_node_names();
+    EXPECT_EQ("/my/ns/my_node1", names_and_namespaces[0]);
+    EXPECT_EQ("/my/ns/my_node2", names_and_namespaces[1]);
+    EXPECT_EQ("/ns2/my_node3", names_and_namespaces[2]);
+    EXPECT_EQ("/my/ns3/my_node4", names_and_namespaces[3]);
```

Comment:
The API doesn't guarantee the order of the nodes in this listing, so you're sort of depending on implementation behavior here. I'd suggest just checking that each of these exist somewhere in the output. This would be robust to reordering and also to other nodes existing at the same time (imagine a new test running node or something).

Reply:
Excellent thinking. Also, didn't realize that about the API. Oops.

I implemented this functionality via an unordered set that's generated from the vector.